From 301ad30f90f239bd48dc7715756f347599a4c84f Mon Sep 17 00:00:00 2001
Message-Id: <301ad30f90f239bd48dc7715756f347599a4c84f.1650636480.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: Hangtian Zhu <hangtian@codeaurora.org>
Date: Tue, 29 Oct 2019 11:36:11 +0800
Subject: [PATCH 02/56] qcacld-2.0: Update regulatory version: 27, 28, 29, 30

Update regulatory version: 27, 28, 29, 30 (2018-11-15 to 2019-9-12)

add country:
ver 29:
CTRY_ALAND_ISLANDS 248 FCC3_WORLD "AX" "Aland Islands"
CTRY_ANDORRA 20 ETSI3_WORLD "AD" "Andorra"
CTRY_ANTIGUA_AND_BARBUDA 28 ETSI13_WORLD "AG" "Antigua and Barbuda"
CTRY_CAMEROON 120 ETSI1_WORLD "CM" "Cameroon"
CTRY_COOK_ISLANDS 184 FCC3_WORLDA "CK" "Cook Islands"
CTRY_FALKLAND_ISLANDS 238 ETSI13_WORLD "FK" "Falk Islands"
CTRY_FAROE_ISLANDS 234 ETSI13_WORLD "FO" "Faroe Islands"
CTRY_FRENCH_SOUTHERN_TERRITORIES 260 ETSI13_WORLD "TF" "French southern
territories"
CTRY_GUERNSEY 831 ETSI13_WORLD "GG" "Guernsey"
CTRY_HEARD_ISLAND_AND_MCDONALD_ISLANDS 334 FCC6_WORLD "HM" "Heard Island
and Mcdonald Islands"
CTRY_HOLY_SEE 336 ETSI13_WORLD "VA" "Holy See"
CTRY_ISLE_OF_MAN 833 ETSI13_WORLD "IM" "Isle of man"
CTRY_JERSEY 832 ETSI13_WORLD "JE" "Jersey"
CTRY_MONTSERRAT 500 ETSI13_WORLD "MS" "Montserrat"
CTRY_NEW_CALEDONIA 540 ETSI13_WORLD "NC" "New Caledonia"
CTRY_NIUE 570 ETSI13_WORLD "NU" "Niue"
CTRY_NORFOLK_ISLAND 574 FCC6_WORLD "NF" "Norfolk Island"
CTRY_SAINT_HELENA_ASCENSION_AND_TRISTAN_DA_CUNHA 654 ETSI13_WORLD "SH"
"Saint Helena Ascension and Tristan DA Cunha"
CTRY_SAN_MARINO 674 FCC3_FCCA "SM" "San Marnino"
CTRY_SAO_TOME_AND_PRINCIPE 678 FCC3_WORLD "ST" "Sao Tome and Principe"
CTRY_SINT_MAARTEN 534 ETSI13_WORLD "SX" "Sint Maarten"
CTRY_SVALBARD_AND_JAN_MAYEN 744 FCC6_WORLD "SJ" "Svalbard and Jan Mayen"
CTRY_UNITED_STATES_MINOR_OUTLYING_ISLANDS 581 FCC3_FCCA "UM"
"United States Minor Outlying Islands"
CTRY_VIRGIN_ISLANDS_BRITISH 92 ETSI13_WORLD "VG" "Virgin Islands British"

change country:
ver 27:
CTRY_CHILE change to FCC13_WORLD
CTRY_COSTA_RICA BW flags to all 'Y'
ver 29:
CTRY_DOMINICAN_REPUBLIC FCC3_FCCA
CTRY_GEORGIA change to ETSI1_WORLD
CTRY_KAZAKHSTAN change to MKK5_MKKC
CTRY_KENYA change to ETSI13_WORLD
ver 30:
CTRY_ARGENTINA change to APL17_ETSIC
CTRY_INDONESIA change to APL2_ETSID

change domain:
ver 27:
APL8 ctl change to FCC
ver 29:
APL9 ctl change to MKK
ver 30:
APL11 DFS change to DFS_FCC3
APL14 ctl change to MKK
APL19 DFS change to DFS_FCC3

Full regdomain defs:
ver 27:
add ETSI10_WORLD ETSI10_FCCA
ver 29:
add APL9_MKKC APL9_KRRA MKK17_MKKC
ver 30:
add APL2_ETSID APL17_ETSIC APL24_ETSIC

add domain:
ver 27:
ETSI10 ETSI DFS_ETSI
ver 29:
KRRA KOR NO_DFS
MKK17 MKK DFS_MKK4
ver 30:
APL24 ETSI DFS_ETSI

Change-Id: I36590a1c234e32ab2e84df33553a2dea39641811
CRs-Fixed: 2554857
---
 CORE/SERVICES/WMA/regdomain.h        |  75 ++++-
 CORE/SERVICES/WMA/regdomain_common.h | 540 ++++++++++++++++++++++++-----------
 2 files changed, 443 insertions(+), 172 deletions(-)

diff --git a/CORE/SERVICES/WMA/regdomain.h b/CORE/SERVICES/WMA/regdomain.h
index 7efcb9d..bfe7f18 100644
--- a/CORE/SERVICES/WMA/regdomain.h
+++ b/CORE/SERVICES/WMA/regdomain.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2014-2018 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2011, 2014-2019 The Linux Foundation. All rights reserved.
  *
  * Previously licensed under the ISC license by Qualcomm Atheros, Inc.
  *
@@ -160,6 +160,8 @@ enum {
                NB: Must agree with macro below (BM) */
 #define BMZERO {(u_int64_t) 0, (u_int64_t) 0}    /* BMLEN zeros */
 
+#define BMNOTZERO {(u_int64_t) 1, (u_int64_t) 0}    /* BMLEN not zeros */
+
 #ifndef SUPPRESS_SHIFT_WARNING
 #define SUPPRESS_SHIFT_WARNING
 #endif
@@ -891,8 +893,14 @@ typedef struct ath_hal_reg_dmn_tables {
  *            - country ID: 4100
  */
 enum CountryCode {
+    CTRY_AFGHANISTAN          = 4,       /* Afghanistan */
+    CTRY_ALAND_ISLANDS        = 248,     /* Aland Islands */
     CTRY_ALBANIA              = 8,       /* Albania */
     CTRY_ALGERIA              = 12,      /* Algeria */
+    CTRY_AMERICAN_SAMOA       = 16,
+    CTRY_ANDORRA              = 20,      /* Andorra */
+    CTRY_ANGUILLA             = 660,
+    CTRY_ANTIGUA_AND_BARBUDA  = 28,      /* Antigua and Barbuda */
     CTRY_ARGENTINA            = 32,      /* Argentina */
     CTRY_ARMENIA              = 51,      /* Armenia */
     CTRY_AUSTRALIA            = 36,      /* Australia */
@@ -906,37 +914,60 @@ enum CountryCode {
     CTRY_BELGIUM              = 56,      /* Belgium */
     CTRY_BELIZE               = 84,      /* Belize */
     CTRY_BERMUDA              = 60,      /* Berumuda */
+    CTRY_BHUTAN               = 64,
     CTRY_BOLIVIA              = 68,      /* Bolivia */
     CTRY_BOSNIA_HERZ          = 70,      /* Bosnia and Herzegowina */
     CTRY_BRAZIL               = 76,      /* Brazil */
     CTRY_BRUNEI_DARUSSALAM    = 96,      /* Brunei Darussalam */
     CTRY_BULGARIA             = 100,     /* Bulgaria */
+    CTRY_BURKINA_FASO         = 854,
     CTRY_CAMBODIA             = 116,     /* Cambodia */
+    CTRY_CAMEROON             = 120,     /* Cameroon */
     CTRY_CANADA               = 124,     /* Canada */
+    CTRY_CAYMAN_ISLANDS       = 136,
+    CTRY_CENTRAL_AFRICA_REPUBLIC               = 140,
+    CTRY_CHAD                 = 148,
     CTRY_CHILE                = 152,     /* Chile */
     CTRY_CHINA                = 156,     /* People's Republic of China */
+    CTRY_CHRISTMAS_ISLAND     = 162,
     CTRY_COLOMBIA             = 170,     /* Colombia */
+    CTRY_COOK_ISLANDS         = 184,     /* Cook Islands */
     CTRY_COSTA_RICA           = 188,     /* Costa Rica */
+    CTRY_COTE_DIVOIRE         = 384,
     CTRY_CROATIA              = 191,     /* Croatia */
     CTRY_CYPRUS               = 196,
     CTRY_CZECH                = 203,     /* Czech Republic */
     CTRY_DENMARK              = 208,     /* Denmark */
+    CTRY_DOMINICA             = 212,
     CTRY_DOMINICAN_REPUBLIC   = 214,     /* Dominican Republic */
     CTRY_ECUADOR              = 218,     /* Ecuador */
     CTRY_EGYPT                = 818,     /* Egypt */
     CTRY_EL_SALVADOR          = 222,     /* El Salvador */
     CTRY_ESTONIA              = 233,     /* Estonia */
+    CTRY_ETHIOPIA              = 231,
+    CTRY_FALKLAND_ISLANDS     = 238,     /* Falkland Islands */
+    CTRY_FAROE_ISLANDS        = 234,     /* Faroe Islands */
     CTRY_FAEROE_ISLANDS       = 234,     /* Faeroe Islands */
     CTRY_FINLAND              = 246,     /* Finland */
     CTRY_FRANCE               = 250,     /* France */
+    CTRY_FRENCH_GUIANA        = 254,
+    CTRY_FRENCH_SOUTHERN_TERRITORIES = 260, /* French southern territories */
+    CTRY_FRENCH_POLYNESIA     = 258,
     CTRY_GEORGIA              = 268,     /* Georgia */
     CTRY_GERMANY              = 276,     /* Germany */
+    CTRY_GHANA                = 288,
+    CTRY_GIBRALTAR            = 292,
     CTRY_GREECE               = 300,     /* Greece */
     CTRY_GREENLAND            = 304,     /* Greenland */
     CTRY_GRENADA              = 308,     /* Grenada */
+    CTRY_GUADELOUPE           = 312,
     CTRY_GUAM                 = 316,     /* Guam */
     CTRY_GUATEMALA            = 320,     /* Guatemala */
+    CTRY_GUERNSEY             = 831,     /* Guernsey */
+    CTRY_GUYANA               = 328,
     CTRY_HAITI                = 332,     /* Haiti */
+    CTRY_HEARD_ISLAND_AND_MCDONALD_ISLANDS = 334,     /* Heard Island and Mcdonald Islands */
+    CTRY_HOLY_SEE             = 336,     /* Holy See */
     CTRY_HONDURAS             = 340,     /* Honduras */
     CTRY_HONG_KONG            = 344,     /* Hong Kong S.A.R., P.R.C. */
     CTRY_HUNGARY              = 348,     /* Hungary */
@@ -946,10 +977,12 @@ enum CountryCode {
     CTRY_IRAN                 = 364,     /* Iran */
     CTRY_IRAQ                 = 368,     /* Iraq */
     CTRY_IRELAND              = 372,     /* Ireland */
+    CTRY_ISLE_OF_MAN          = 833,     /* Isle of man */
     CTRY_ISRAEL               = 376,     /* Israel */
     CTRY_ITALY                = 380,     /* Italy */
     CTRY_JAMAICA              = 388,     /* Jamaica */
     CTRY_JAPAN                = 392,     /* Japan */
+    CTRY_JERSEY               = 832,     /* Jersey */
     CTRY_JORDAN               = 400,     /* Jordan */
     CTRY_KAZAKHSTAN           = 398,     /* Kazakhstan */
     CTRY_KENYA                = 404,     /* Kenya */
@@ -959,28 +992,45 @@ enum CountryCode {
     CTRY_KUWAIT               = 414,     /* Kuwait */
     CTRY_LATVIA               = 428,     /* Latvia */
     CTRY_LEBANON              = 422,     /* Lebanon */
+    CTRY_LESOTHO              = 426,
     CTRY_LIBYA                = 434,     /* Libya */
     CTRY_LIECHTENSTEIN        = 438,     /* Liechtenstein */
     CTRY_LITHUANIA            = 440,     /* Lithuania */
     CTRY_LUXEMBOURG           = 442,     /* Luxembourg */
     CTRY_MACAU                = 446,     /* Macau SAR */
     CTRY_MACEDONIA            = 807,     /* the Former Yugoslav Republic of Macedonia */
+    CTRY_MALAWI               = 454,
     CTRY_MALAYSIA             = 458,     /* Malaysia */
     CTRY_MALDIVES             = 462,     /* Maldives */
     CTRY_MALTA                = 470,     /* Malta */
+    CTRY_MARSHALL_ISLANDS     = 584,
+    CTRY_MARTINIQUE           = 474,
+    CTRY_MAURITANIA           = 478,
     CTRY_MAURITIUS            = 480,     /* Mauritius */
+    CTRY_MAYOTTE              = 175,
     CTRY_MEXICO               = 484,     /* Mexico */
+    CTRY_MICRONESIA           = 583,
+    CTRY_MOLDOVA              = 498,
     CTRY_MONACO               = 492,     /* Principality of Monaco */
+    CTRY_MONGOLIA             = 496,
     CTRY_MOROCCO              = 504,     /* Morocco */
+    CTRY_MYANMAR              = 104,     /* Myanmar*/
+    CTRY_NAMIBIA              = 516,
     CTRY_NEPAL                = 524,     /* Nepal */
     CTRY_NETHERLANDS          = 528,     /* Netherlands */
     CTRY_NETHERLANDS_ANTILLES = 530,     /* Netherlands-Antilles */
+    CTRY_NEW_CALEDONIA        = 540,     /* New Caledonia */
     CTRY_ARUBA                = 533,     /* Aruba */
     CTRY_NEW_ZEALAND          = 554,     /* New Zealand */
+    CTRY_NIGERIA              = 566,
+    CTRY_NIUE                 = 570,     /* Niue */
+    CTRY_NORFOLK_ISLAND       = 574,     /* Norfolk Island */
+    CTRY_NORTHERN_MARIANA_ISLANDS          = 580,
     CTRY_NICARAGUA            = 558,     /* Nicaragua */
     CTRY_NORWAY               = 578,     /* Norway */
     CTRY_OMAN                 = 512,     /* Oman */
     CTRY_PAKISTAN             = 586,     /* Islamic Republic of Pakistan */
+    CTRY_PALAU                = 585,
     CTRY_PANAMA               = 591,     /* Panama */
     CTRY_PAPUA_NEW_GUINEA     = 598,     /* Papua New Guinea */
     CTRY_PARAGUAY             = 600,     /* Paraguay */
@@ -990,17 +1040,33 @@ enum CountryCode {
     CTRY_PORTUGAL             = 620,     /* Portugal */
     CTRY_PUERTO_RICO          = 630,     /* Puerto Rico */
     CTRY_QATAR                = 634,     /* Qatar */
+    CTRY_REUNION              = 638,
     CTRY_ROMANIA              = 642,     /* Romania */
     CTRY_RUSSIA               = 643,     /* Russia */
     CTRY_RWANDA               = 646,     /* Rwanda */
+    CTRY_SAINT_BARTHELEMY     = 652,
+    CTRY_SAINT_HELENA_ASCENSION_AND_TRISTAN_DA_CUNHA = 654, /* Saint Helena Ascension and Tristan DA Cunha */
+    CTRY_SAINT_KITTS_AND_NEVIS               = 659,
+    CTRY_SAINT_LUCIA          = 662,
+    CTRY_SAINT_MARTIN         = 663,
+    CTRY_SAINT_PIERRE_AND_MIQUELON               = 666,
+    CTRY_SAINT_VINCENT_AND_THE_GRENADIENS               = 670,
+    CTRY_SAMOA               = 882,
+    CTRY_SAN_MARINO          = 674,      /* San Marnino */
+    CTRY_SAO_TOME_AND_PRINCIPE = 678,    /* Sao Tome and Principe */
     CTRY_SAUDI_ARABIA         = 682,     /* Saudi Arabia */
+    CTRY_SENEGAL              = 686,
     CTRY_SERBIA               = 688,     /* Republic of Serbia */
     CTRY_MONTENEGRO           = 499,     /* Montenegro */
+    CTRY_MONTSERRAT           = 500,     /* Montserrat */
     CTRY_SINGAPORE            = 702,     /* Singapore */
+    CTRY_SINT_MAARTEN         = 534,     /* Sint Maarten */
     CTRY_SLOVAKIA             = 703,     /* Slovak Republic */
     CTRY_SLOVENIA             = 705,     /* Slovenia */
     CTRY_SOUTH_AFRICA         = 710,     /* South Africa */
     CTRY_SPAIN                = 724,     /* Spain */
+    CTRY_SURINAME                = 740,
+    CTRY_SVALBARD_AND_JAN_MAYEN  = 744, /* Svalbard and Jan Mayen */
     CTRY_SRI_LANKA            = 144,     /* Sri Lanka */
     CTRY_SWEDEN               = 752,     /* Sweden */
     CTRY_SWITZERLAND          = 756,     /* Switzerland */
@@ -1008,20 +1074,27 @@ enum CountryCode {
     CTRY_TAIWAN               = 158,     /* Taiwan */
     CTRY_TANZANIA             = 834,     /* Tanzania */
     CTRY_THAILAND             = 764,     /* Thailand */
+    CTRY_TOGO                 = 768,
     CTRY_TRINIDAD_Y_TOBAGO    = 780,     /* Trinidad y Tobago */
     CTRY_TUNISIA              = 788,     /* Tunisia */
     CTRY_TURKEY               = 792,     /* Turkey */
+    CTRY_TURKS_AND_CAICOS     = 796,
     CTRY_UAE                  = 784,     /* U.A.E. */
     CTRY_UGANDA               = 800,     /* Uganda */
     CTRY_UKRAINE              = 804,     /* Ukraine */
     CTRY_UNITED_KINGDOM       = 826,     /* United Kingdom */
     CTRY_UNITED_STATES        = 840,     /* United States */
+    CTRY_UNITED_STATES_MINOR_OUTLYING_ISLANDS = 581,     /* United States Minor Outlying Islands */
     CTRY_UNITED_STATES2       = 841,     /* United States for AP */
     CTRY_UNITED_STATES_FCC49  = 842,     /* United States (Public Safety)*/
     CTRY_URUGUAY              = 858,     /* Uruguay */
     CTRY_UZBEKISTAN           = 860,     /* Uzbekistan */
+    CTRY_VANUATU              = 548,
     CTRY_VENEZUELA            = 862,     /* Venezuela */
     CTRY_VIET_NAM             = 704,     /* Viet Nam */
+    CTRY_VIRGIN_ISLANDS_BRITISH     = 92, /* Virgin Islands British */
+    CTRY_VIRGIN_ISLANDS       = 850,
+    CTRY_WALLIS_AND_FUTUNA    = 876,
     CTRY_YEMEN                = 887,     /* Yemen */
     CTRY_ZIMBABWE             = 716,     /* Zimbabwe */
 
diff --git a/CORE/SERVICES/WMA/regdomain_common.h b/CORE/SERVICES/WMA/regdomain_common.h
index 862e684..aa87f1e 100644
--- a/CORE/SERVICES/WMA/regdomain_common.h
+++ b/CORE/SERVICES/WMA/regdomain_common.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2014-2016 2018 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2011, 2014-2016 2018-2019 The Linux Foundation. All rights reserved.
  *
  * Previously licensed under the ISC license by Qualcomm Atheros, Inc.
  *
@@ -102,6 +102,7 @@ enum EnumRd {
     FCC4_FCCA   = 0x12,     /* USA - Public Safety */
     FCC5_FCCA   = 0x13,     /* US with no DFS (UNII-1 + UNII-3 Only)*/
     FCC6_FCCA   = 0x14,     /* Canada for AP only*/
+    FCC8_FCCA   = 0x16,
 
     FCC2_FCCA   = 0x20,     /* Canada */
     FCC2_WORLD  = 0x21,     /* Australia & HK */
@@ -111,6 +112,9 @@ enum EnumRd {
     FCC3_FCCA   = 0x3A,     /* USA & Canada w/5470 band, 11h, DFS enabled */
     FCC3_WORLD  = 0x3B,     /* USA & Canada w/5470 band, 11h, DFS enabled */
     FCC3_ETSIC  = 0x3F,     /* New Zealand, DFS enabled */
+    FCC11_WORLD = 0x19,
+    FCC13_WORLD = 0xE4,
+    FCC14_FCCB = 0xE6,
 
     ETSI1_WORLD = 0x37,
     ETSI3_ETSIA = 0x32,     /* France (optional) */
@@ -122,6 +126,11 @@ enum EnumRd {
     ETSI6_WORLD = 0x34,     /* Bulgaria */
     ETSI8_WORLD = 0x3D,     /* Russia */
     ETSI9_WORLD = 0x3E,     /* Ukraine */
+    ETSI10_WORLD = 0x24,    /* ETSI1 STA and AP with DSRC */
+    ETSI10_FCCA = 0x25,     /* Same as above but FCCA in 2G */
+    ETSI13_WORLD = 0x27,
+    ETSI14_WORLD = 0x29,
+    ETSI15_WORLD = 0x31,
     ETSI_RESERVED   = 0x33,     /* Reserved (Do not used) */
 
     MKK1_MKKA   = 0x40,     /* Japan (JP1) */
@@ -147,13 +156,24 @@ enum EnumRd {
     APL1_APLA   = 0x54,
     APL1_ETSIC  = 0x55,
     APL2_ETSIC  = 0x56,     /* Venezuela */
+    APL2_ETSID  = 0x41,     /* Indonesia Outdoor */
     APL5_WORLD  = 0x58,     /* Chile */
     APL13_WORLD = 0x5A,     /* Algeria */
     APL6_WORLD  = 0x5B,     /* Singapore */
     APL7_FCCA   = 0x5C,     /* Taiwan 5.47 Band */
     APL8_WORLD  = 0x5D,     /* Malaysia 5GHz */
-    APL9_MKKC   = 0x5E,     /* Korea 5GHz, Before 11/2007. Now used only by APs */
-    APL10_MKKC  = 0x5F,     /* Korea 5GHz, After 11/2007. For STAs only */
+    APL9_WORLD   = 0x5E,
+    APL9_MKKC   = 0x48,
+    APL9_KRRA   = 0x43,
+    APL10_WORLD = 0x5F,     /* Korea 5GHz, After 11/2007. For STAs only */
+    APL17_ETSID = 0xE0,
+    APL17_ETSIC = 0xE7,
+    APL14_WORLD = 0x57,
+    APL15_WORLD = 0x59,
+    APL19_ETSIC = 0x71,
+    APL20_WORLD = 0xE5,
+    APL23_WORLD = 0xE3,
+    APL24_ETSIC = 0xE2,
 
     /*
      * World mode SKUs
@@ -220,6 +240,7 @@ enum EnumRd {
     MKK12_MKKA2 = 0xDD,     /* Japan UNI-1 even + UNI-1 odd + UNI-2 + mid-band + 4.9GHz + MKKA2 */
 
     MKK13_MKKB  = 0xDE,     /* Japan UNI-1 even + UNI-1 odd + UNI-2 + mid-band + MKKB + All passive + no adhoc */
+    MKK17_MKKC  = 0xE1,
 
     /* Following definitions are used only by s/w to map old
      * Japan SKUs.
@@ -261,6 +282,13 @@ enum EnumRd {
     APL11       = 0x1150,   /* Specific AP Customer 5GHz, For APs Only */
     APL12       = 0x1160,   /* Kenya */
     APL13       = 0x1170,   /* Algeria */
+    APL14       = 0x1180,
+    APL15       = 0x1190,
+    APL17       = 0x1210,
+    APL19       = 0x1240,
+    APL20       = 0x1250,
+    APL23       = 0x1280,
+    APL24       = 0x1290,
 
     ETSI1       = 0x0130,   /* Europe & others */
     ETSI2       = 0x0230,   /* Europe & others */
@@ -273,6 +301,11 @@ enum EnumRd {
     ETSIA       = 0x0A30,   /* France */
     ETSIB       = 0x0B30,   /* Israel */
     ETSIC       = 0x0C30,   /* Latin America */
+    ETSID       = 0x0F30,
+    ETSI10      = 0x0D30,
+    ETSI13     = 0x0E39,
+    ETSI14     = 0x0E40,
+    ETSI15     = 0x0E41,
 
     FCC1        = 0x0110,   /* US & others */
     FCC2        = 0x0120,   /* Canada, Australia & New Zealand */
@@ -280,7 +313,13 @@ enum EnumRd {
     FCC4        = 0x0165,   /* US Public Safety */
     FCC5        = 0x0510,
     FCC6        = 0x0610,   /* Canada & Australia */
+    FCC8        = 0x0810,
+    FCC11     	= 0x0B20,
+    FCC13     	= 0x0B60,
+    FCC14     	= 0x0B70,
     FCCA        = 0x0A10,
+    FCCB        = 0x0B90,
+    KRRA        = 0x0A60,
 
     APLD        = 0x0D50,   /* South Korea */
 
@@ -299,6 +338,7 @@ enum EnumRd {
     MKK13       = 0x0C40,   /* Same as MKK8 but all passive and no adhoc 11a */
     MKK14       = 0x1440,   /* Japan UNI-1 even + UNI-1 odd + 4.9GHz */
     MKK15       = 0x1540,   /* Japan UNI-1 even + UNI-1 odd + UNI-2 + 4.9GHz */
+    MKK17       = 0x1650,
     MKKA        = 0x0A40,   /* Japan */
     MKKC        = 0x0A50,
 
@@ -311,6 +351,9 @@ enum {					/* conformance test limits */
 	FCC	= 0x10,
 	MKK	= 0x40,
 	ETSI	= 0x30,
+	CHN	= 0x60,
+	KOR	= 0x50,
+	USER_DEF = 0x70,
 };
 /*
  * The following are flags for different requirements per reg domain.
@@ -361,6 +404,11 @@ static const REG_DMN_PAIR_MAPPING ahCmnRegDomainPairs[] = {
 	{ETSI6_WORLD,	ETSI6,		WORLD,		DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB, NO_REQ, PSCAN_DEFER, 0 },
     {ETSI8_WORLD,	ETSI8,		WORLD,		DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB, NO_REQ, PSCAN_DEFER, 0 },
 	{ETSI9_WORLD,	ETSI9,		WORLD,		DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB, NO_REQ, PSCAN_DEFER, 0 },
+	{ETSI10_WORLD,	ETSI10,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{ETSI10_FCCA,	ETSI10,		FCCA,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{ETSI13_WORLD,	ETSI13,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{ETSI14_WORLD,	ETSI14,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{ETSI15_WORLD,	ETSI15,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 
 	{ETSI3_ETSIA,	ETSI3,		WORLD,		DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB, NO_REQ, PSCAN_DEFER, 0 },
 	{FRANCE_RES,	ETSI3,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
@@ -377,11 +425,23 @@ static const REG_DMN_PAIR_MAPPING ahCmnRegDomainPairs[] = {
 	{APL6_WORLD,	APL6,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL7_FCCA,	    APL7,		FCCA,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL8_WORLD,	APL8,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL9_WORLD,	APL9,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL9_MKKC,	APL9,		MKKC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
-	{APL10_MKKC,	APL10,		MKKC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL9_KRRA,	APL9,		KRRA,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL10_WORLD,	APL10,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL12_WORLD,	APL12,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL14_WORLD,	APL14,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL15_WORLD,	APL15,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL19_ETSIC,	APL19,		ETSIC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL20_WORLD,	APL20,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL23_WORLD,	APL23,		WORLD,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL24_ETSIC,	APL24,		ETSIC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL17_ETSID,	APL17,		ETSID,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL17_ETSIC,	APL17,		ETSIC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL3_FCCA,	APL3,		FCCA,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL1_ETSIC,	APL1,		ETSIC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 	{APL2_ETSIC,	APL2,		ETSIC,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
+	{APL2_ETSID,	APL2,		ETSID,		NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
 
 	{MKK1_MKKA,	MKK1,		MKKA,		DISALLOW_ADHOC_11A_TURB | NEED_NFC | LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK1 | PSCAN_MKKA, CTRY_JAPAN },
 	{MKK1_MKKB,	MKK1,		MKKA,		DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB | NEED_NFC| LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK1 | PSCAN_MKKA | PSCAN_MKKA_G, CTRY_JAPAN1 },
@@ -464,6 +524,7 @@ static const REG_DMN_PAIR_MAPPING ahCmnRegDomainPairs[] = {
 	{MKK14_MKKA1,   MKK14,      MKKA,       DISALLOW_ADHOC_11A_TURB | NEED_NFC | LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK1 | PSCAN_MKKA1 | PSCAN_MKKA1_G, CTRY_JAPAN58 },
 	{MKK15_MKKA1,   MKK15,      MKKA,       DISALLOW_ADHOC_11A_TURB | NEED_NFC | LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK1 | PSCAN_MKK3 | PSCAN_MKKA1 | PSCAN_MKKA1_G, CTRY_JAPAN59 },
 	{MKK5_MKKA2,    MKK5,       MKKA,       DISALLOW_ADHOC_11A_TURB | NEED_NFC | LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK3 | PSCAN_MKKA2 | PSCAN_MKKA2_G, CTRY_XA },
+	{MKK17_MKKC,    MKK17,       MKKC,       DISALLOW_ADHOC_11A_TURB | NEED_NFC | LIMIT_FRAME_4MS, NEED_NFC, PSCAN_MKK3 | PSCAN_MKKA2 | PSCAN_MKKA2_G, CTRY_XA },
 
 		/* These are super domains */
 	{WOR0_WORLD,	WOR0_WORLD,	WOR0_WORLD,	NO_REQ, NO_REQ, PSCAN_DEFER, 0 },
@@ -483,14 +544,19 @@ static const REG_DMN_PAIR_MAPPING ahCmnRegDomainPairs[] = {
 
 static const COUNTRY_CODE_TO_ENUM_RD ahCmnAllCountries[] = {
     {CTRY_DEBUG,       NO_ENUMRD,     "DB", "DEBUG",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ALBANIA,     ETSI1_WORLD,   "AL", "ALBANIA",        YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
-    {CTRY_ALGERIA,     APL13_WORLD,   "DZ", "ALGERIA",        YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
-    {CTRY_ARGENTINA,   FCC3_WORLD,    "AR", "ARGENTINA",      YES,  NO,  NO, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ARMENIA,     ETSI4_WORLD,   "AM", "ARMENIA",        YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_ARUBA,       ETSI1_WORLD,   "AW", "ARUBA",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_AUSTRALIA,   FCC3_WORLD,    "AU", "AUSTRALIA",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_AUSTRALIA2,  FCC6_WORLD,    "AU", "AUSTRALIA2",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_AUSTRIA,     ETSI1_WORLD,   "AT", "AUSTRIA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_AFGHANISTAN,     ETSI1_WORLD,   "AF", "AFGHANISTAN",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ALAND_ISLANDS,     FCC3_WORLD,   "AX", "ALAND ISLANDS",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ALBANIA,     ETSI13_WORLD,   "AL", "ALBANIA",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ALGERIA,     APL13_WORLD,   "DZ", "ALGERIA",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_AMERICAN_SAMOA,     FCC3_FCCA,   "AS", "AMERICAN SAMOA",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ANDORRA,     ETSI3_WORLD,   "AD", "ANDORRA",        YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ANGUILLA,     ETSI1_WORLD,   "AI", "ANGUILLA",	  YES, YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ANTIGUA_AND_BARBUDA,     ETSI13_WORLD,   "AG", "ANTIGUA AND BARBUDA",	  YES, YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_ARGENTINA,   APL17_ETSIC,    "AR", "ARGENTINA",      YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ARMENIA,     ETSI4_WORLD,   "AM", "ARMENIA",        YES,  YES, YES, YES, YES, YES,  NO, NO, 7000 },
+    {CTRY_ARUBA,       ETSI1_WORLD,   "AW", "ARUBA",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_AUSTRALIA,   FCC6_WORLD,    "AU", "AUSTRALIA",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_AUSTRIA,     ETSI13_WORLD,   "AT", "AUSTRIA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_AZERBAIJAN,  ETSI4_WORLD,   "AZ", "AZERBAIJAN",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_BAHAMAS,     FCC3_WORLD,    "BS", "BAHAMAS",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_BAHRAIN,     APL6_WORLD,    "BH", "BAHRAIN",        YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
@@ -505,129 +571,109 @@ static const COUNTRY_CODE_TO_ENUM_RD ahCmnAllCountries[] = {
     {CTRY_BOSNIA_HERZ, ETSI1_WORLD,   "BA", "BOSNIA AND HERZEGOVINA", YES,  NO, YES, YES, YES, YES,  YES, YES, 7000 },
     {CTRY_BRAZIL,      FCC3_WORLD,    "BR", "BRAZIL",         YES,  NO,  NO, YES, YES, YES, YES, YES, 7000 },
     {CTRY_BRUNEI_DARUSSALAM, APL6_WORLD, "BN", "BRUNEI DARUSSALAM", YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_BULGARIA,    ETSI1_WORLD,   "BG", "BULGARIA",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CAMBODIA,    ETSI1_WORLD,   "KH", "CAMBODIA",       YES, NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CANADA,      FCC3_FCCA,     "CA", "CANADA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CANADA2,     FCC6_FCCA,     "CA", "CANADA2",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CHILE,       APL6_WORLD,    "CL", "CHILE",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CHINA,       APL1_WORLD,    "CN", "CHINA",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_COLOMBIA,    FCC3_WORLD,    "CO", "COLOMBIA",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_COSTA_RICA,  FCC1_WORLD,    "CR", "COSTA RICA",     YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_CROATIA,     ETSI1_WORLD,   "HR", "CROATIA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CYPRUS,      ETSI1_WORLD,   "CY", "CYPRUS",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_CZECH,       ETSI1_WORLD,   "CZ", "CZECH REPUBLIC", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_DENMARK,     ETSI1_WORLD,   "DK", "DENMARK",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_DOMINICAN_REPUBLIC, FCC1_FCCA, "DO", "DOMINICAN REPUBLIC", YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ECUADOR,     FCC1_WORLD,    "EC", "ECUADOR",        YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_EGYPT,       ETSI3_WORLD,   "EG", "EGYPT",          YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_EL_SALVADOR, FCC1_WORLD,    "SV", "EL SALVADOR",    YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_ESTONIA,     ETSI1_WORLD,   "EE", "ESTONIA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_FINLAND,     ETSI1_WORLD,   "FI", "FINLAND",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_FRANCE,      ETSI1_WORLD,   "FR", "FRANCE",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_GEORGIA,     ETSI4_WORLD,   "GE", "GEORGIA",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_GERMANY,     ETSI1_WORLD,   "DE", "GERMANY",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_GREECE,      ETSI1_WORLD,   "GR", "GREECE",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_GREENLAND,   ETSI1_WORLD,   "GL", "GREENLAND",      YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_GRENADA,     FCC3_FCCA,     "GD", "GRENADA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_GUAM,        FCC1_FCCA,     "GU", "GUAM",           YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_GUATEMALA,   FCC1_FCCA,     "GT", "GUATEMALA",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_HAITI,       ETSI1_WORLD,   "HT", "HAITI",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_HONDURAS,    FCC3_WORLD,    "HN", "HONDURAS",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_BULGARIA,    ETSI13_WORLD,   "BG", "BULGARIA",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_BURKINA_FASO,    FCC3_WORLD,   "BF", "BURKINA-FASO",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CAMBODIA,    ETSI1_WORLD,   "KH", "CAMBODIA",       YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CAMEROON,    ETSI1_WORLD,   "CM", "CAMEROON",       YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CANADA,      FCC6_FCCA,     "CA", "CANADA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CAYMAN_ISLANDS,       FCC3_WORLD,    "KY", "CAYMAN ISLANDS",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CENTRAL_AFRICA_REPUBLIC,	FCC3_WORLD,    "CF", "AFRICA REPUBLIC",        YES, YES, YES, YES, YES, YES, YES, NO, 7000 },
+    {CTRY_CHAD,       ETSI1_WORLD,    "TD", "CHAD",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CHILE,       FCC13_WORLD,    "CL", "CHILE",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CHINA,       APL14_WORLD,    "CN", "CHINA",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CHRISTMAS_ISLAND, FCC3_WORLD, "CX", "CHRISTMAS ISLAND",          YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_COLOMBIA,    FCC3_WORLD,    "CO", "COLOMBIA",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_COOK_ISLANDS,    FCC3_WORLD,    "CK", "COOK ISLANDS",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_COSTA_RICA,  FCC3_WORLD,    "CR", "COSTA RICA",     YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_COTE_DIVOIRE, FCC3_WORLD, "CI", "COTE DIVOIRE",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CROATIA,     ETSI13_WORLD,   "HR", "CROATIA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CYPRUS,      ETSI13_WORLD,   "CY", "CYPRUS",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_CZECH,       ETSI13_WORLD,   "CZ", "CZECH REPUBLIC", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_DENMARK,     ETSI13_WORLD,   "DK", "DENMARK",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_DOMINICA, FCC2_FCCA, "DM", "DOMINICA",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_DOMINICAN_REPUBLIC, FCC3_FCCA, "DO", "DOMINICAN REPUBLIC", YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ECUADOR,     FCC3_WORLD,    "EC", "ECUADOR",        YES,  YES,  YES, YES, YES, YES,  NO, NO, 7000 },
+    {CTRY_EGYPT,       ETSI3_WORLD,   "EG", "EGYPT",          YES,  NO, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_EL_SALVADOR, FCC2_WORLD,    "SV", "EL SALVADOR",    YES,  YES, YES, YES, YES, YES,  NO, NO, 7000 },
+    {CTRY_ESTONIA,     ETSI13_WORLD,   "EE", "ESTONIA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ETHIOPIA, ETSI1_WORLD, "ET", "ETHIOPIA",   YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FALKLAND_ISLANDS, ETSI13_WORLD, "FK", "FALKLAND ISLANDS",   YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FAROE_ISLANDS, ETSI13_WORLD, "FO", "FAROE ISLANDS",   YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FINLAND,     ETSI13_WORLD,   "FI", "FINLAND",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FRANCE,      ETSI13_WORLD,   "FR", "FRANCE",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FRENCH_GUIANA, ETSI13_WORLD, "GF", "FRENCH GUIANA",           YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FRENCH_SOUTHERN_TERRITORIES, ETSI13_WORLD, "TF", "FRENCH SOURTHERN TERRITORIES",           YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_FRENCH_POLYNESIA, ETSI13_WORLD, "PF", "FRENCH POLYNESIA",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GEORGIA,     ETSI1_WORLD,   "GE", "GEORGIA",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GERMANY,     ETSI13_WORLD,   "DE", "GERMANY",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GHANA, FCC3_WORLD, "GH", "GHANA",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GIBRALTAR, ETSI1_WORLD, "GI", "GIBRALTAR",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GREECE,      ETSI13_WORLD,   "GR", "GREECE",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GREENLAND,   ETSI1_WORLD,   "GL", "GREENLAND",      YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_GRENADA,     FCC3_FCCA,     "GD", "GRENADA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GUADELOUPE, ETSI1_WORLD, "GP", "GUADELOUPE",        YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GUAM,        FCC3_FCCA,     "GU", "GUAM",           YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_GUATEMALA,   ETSI1_WORLD,     "GT", "GUATEMALA",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GUERNSEY,   ETSI13_WORLD,     "GG", "GUERNSEY",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_GUYANA, APL1_ETSIC, "GY", "GUYANA",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_HAITI,       FCC3_FCCA,   "HT", "HAITI",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_HEARD_ISLAND_AND_MCDONALD_ISLANDS,    FCC6_WORLD,    "HM", "HEARD ISLAND AND MCDONALD ISLANDS",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_HOLY_SEE,    ETSI13_WORLD,    "VA", "HOLY SEE",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_HONDURAS,    FCC13_WORLD,    "HN", "HONDURAS",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_HONG_KONG,   FCC3_WORLD,    "HK", "HONG KONG",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_HUNGARY,     ETSI1_WORLD,   "HU", "HUNGARY",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ICELAND,     ETSI1_WORLD,   "IS", "ICELAND",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_INDIA,       APL6_WORLD,    "IN", "INDIA",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_INDONESIA,   APL2_WORLD,    "ID", "INDONESIA",      YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_IRAN,        APL1_WORLD,    "IR", "IRAN",           YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_IRELAND,     ETSI1_WORLD,   "IE", "IRELAND",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ISRAEL,      ETSI3_WORLD,   "IL", "ISRAEL",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ITALY,       ETSI1_WORLD,   "IT", "ITALY",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_JAMAICA,     FCC3_WORLD,    "JM", "JAMAICA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_JAPAN,       MKK5_MKKA2,    "JP", "JAPAN",          YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN1,      MKK1_MKKB,     "JP", "JAPAN1",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN2,      MKK1_FCCA,     "JP", "JAPAN2",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN3,      MKK2_MKKA,     "JP", "JAPAN3",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN4,      MKK1_MKKA1,    "JP", "JAPAN4",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN5,      MKK1_MKKA2,    "JP", "JAPAN5",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN6,      MKK1_MKKC,     "JP", "JAPAN6",         YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JAPAN7,      MKK3_MKKB,     "JP", "JAPAN7",         YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN8,      MKK3_MKKA2,    "JP", "JAPAN8",         YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN9,      MKK3_MKKC,     "JP", "JAPAN9",         YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN10,     MKK4_MKKB,     "JP", "JAPAN10",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN11,     MKK4_MKKA2,    "JP", "JAPAN11",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN12,     MKK4_MKKC,     "JP", "JAPAN12",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN13,     MKK5_MKKB,     "JP", "JAPAN13",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN14,     MKK5_MKKA2,    "JP", "JAPAN14",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN15,     MKK5_MKKC,     "JP", "JAPAN15",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN16,     MKK6_MKKB,     "JP", "JAPAN16",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN17,     MKK6_MKKA2,    "JP", "JAPAN17",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN18,     MKK6_MKKC,     "JP", "JAPAN18",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN19,     MKK7_MKKB,     "JP", "JAPAN19",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN20,     MKK7_MKKA2,    "JP", "JAPAN20",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN21,     MKK7_MKKC,     "JP", "JAPAN21",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN22,     MKK8_MKKB,     "JP", "JAPAN22",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN23,     MKK8_MKKA2,    "JP", "JAPAN23",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN24,     MKK8_MKKC,     "JP", "JAPAN24",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN25,     MKK3_MKKA,     "JP", "JAPAN25",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN26,     MKK3_MKKA1,    "JP", "JAPAN26",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN27,     MKK3_FCCA,     "JP", "JAPAN27",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN28,     MKK4_MKKA1,    "JP", "JAPAN28",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN29,     MKK4_FCCA,     "JP", "JAPAN29",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN30,     MKK6_MKKA1,    "JP", "JAPAN30",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN31,     MKK6_FCCA,     "JP", "JAPAN31",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN32,     MKK7_MKKA1,    "JP", "JAPAN32",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN33,     MKK7_FCCA,     "JP", "JAPAN33",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN34,     MKK9_MKKA,     "JP", "JAPAN34",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN35,     MKK10_MKKA,    "JP", "JAPAN35",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN36,     MKK4_MKKA,     "JP", "JAPAN36",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN37,     MKK9_FCCA,     "JP", "JAPAN37",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN38,     MKK9_MKKA1,    "JP", "JAPAN38",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN39,     MKK9_MKKC,     "JP", "JAPAN39",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN40,     MKK9_MKKA2,    "JP", "JAPAN40",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN41,     MKK10_FCCA,    "JP", "JAPAN41",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN42,     MKK10_MKKA1,   "JP", "JAPAN42",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN43,     MKK10_MKKC,    "JP", "JAPAN43",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN44,     MKK10_MKKA2,   "JP", "JAPAN44",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN45,     MKK11_MKKA,    "JP", "JAPAN45",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN46,     MKK11_FCCA,    "JP", "JAPAN46",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN47,     MKK11_MKKA1,   "JP", "JAPAN47",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN48,     MKK11_MKKC,    "JP", "JAPAN48",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN49,     MKK11_MKKA2,   "JP", "JAPAN49",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN50,     MKK12_MKKA,    "JP", "JAPAN50",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN51,     MKK12_FCCA,    "JP", "JAPAN51",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN52,     MKK12_MKKA1,   "JP", "JAPAN52",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN53,     MKK12_MKKC,    "JP", "JAPAN53",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN54,     MKK12_MKKA2,   "JP", "JAPAN54",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-/*    {CTRY_JAPAN55,     MKK5_MKKA,     "JP", "JAPAN55",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },*/
-    {CTRY_JAPAN56,     MKK5_FCCA,     "JP", "JAPAN56",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN57,     MKK13_MKKB,    "JP", "JAPAN57",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN58,     MKK14_MKKA1,   "JP", "JAPAN58",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_JAPAN59,     MKK15_MKKA1,   "JP", "JAPAN59",        YES,  NO,  NO, YES, YES, YES, YES, NO, 7000 },
-    {CTRY_XA,          MKK5_MKKA2,    "XA", "JAPAN PASSIVE",  YES,  NO,  NO, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_JORDAN,      ETSI2_WORLD,   "JO", "JORDAN",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_KAZAKHSTAN,  NULL1_WORLD,   "KZ", "KAZAKHSTAN",     YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
-    {CTRY_KENYA,       APL1_WORLD,    "KE", "KENYA",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_KOREA_NORTH, APL9_MKKC,     "KP", "NORTH KOREA",    YES,  NO,  NO, YES,  NO, YES,  NO, NO, 7000 },
-    {CTRY_KOREA_ROC,   APL10_MKKC,    "KR", "KOREA REPUBLIC", YES,  NO,  NO, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_KOREA_ROC3,  APL9_MKKC,     "KR", "KOREA REPUBLIC3", YES,  NO,  NO, YES,  NO, YES,  NO, NO, 7000 },
-    {CTRY_KUWAIT,      ETSI3_WORLD,   "KW", "KUWAIT",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_LATVIA,      ETSI1_WORLD,   "LV", "LATVIA",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_LEBANON,     APL1_WORLD,    "LB", "LEBANON",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_LIECHTENSTEIN, ETSI1_WORLD, "LI", "LIECHTENSTEIN",  YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_LITHUANIA,   ETSI1_WORLD,   "LT", "LITHUANIA",      YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_LUXEMBOURG,  ETSI1_WORLD,   "LU", "LUXEMBOURG",     YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MACAU,       FCC2_WORLD,    "MO", "MACAU SAR",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MACEDONIA,   ETSI1_WORLD,   "MK", "MACEDONIA, FYRO", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MALAYSIA,    FCC1_WORLD,    "MY", "MALAYSIA",       YES,  NO,  NO, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MALTA,       ETSI1_WORLD,   "MT", "MALTA",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MAURITIUS,   ETSI1_WORLD,   "MU", "MAURITIUS",      YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MEXICO,      FCC1_WORLD,    "MX", "MEXICO",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MONACO,      ETSI4_WORLD,   "MC", "MONACO",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MOROCCO,     APL4_WORLD,    "MA", "MOROCCO",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_NEPAL,       APL1_WORLD,    "NP", "NEPAL",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_NETHERLANDS, ETSI1_WORLD,   "NL", "NETHERLANDS",    YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_NETHERLANDS_ANTILLES, ETSI1_WORLD, "AN", "NETHERLANDS ANTILLES", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_NEW_ZEALAND, FCC3_ETSIC,    "NZ", "NEW ZEALAND",    YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_HUNGARY,     ETSI13_WORLD,   "HU", "HUNGARY",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ICELAND,     ETSI13_WORLD,   "IS", "ICELAND",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_INDIA,       APL19_ETSIC,    "IN", "INDIA",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_INDONESIA,   APL2_ETSID,    "ID", "INDONESIA",      YES,  YES, YES, YES, YES, YES,  NO, NO, 7000 },
+    {CTRY_IRAQ, ETSI1_WORLD, "IQ", "IRAQ",       YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_IRELAND,     ETSI13_WORLD,   "IE", "IRELAND",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ISLE_OF_MAN,     ETSI13_WORLD,   "IM", "ISLE OF MAN",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ISRAEL,      ETSI3_WORLD,   "IL", "ISRAEL",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ITALY,       ETSI13_WORLD,   "IT", "ITALY",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_JAMAICA,     FCC13_WORLD,    "JM", "JAMAICA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_JAPAN,       MKK5_MKKC,    "JP", "JAPAN",          YES,  YES,  YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_JERSEY,       ETSI13_WORLD,    "JE", "JERSEY",          YES,  YES,  YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_JORDAN,      APL4_WORLD,   "JO", "JORDAN",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_KAZAKHSTAN,  MKK5_MKKC,   "KZ", "KAZAKHSTAN",     YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
+    {CTRY_KENYA,       ETSI13_WORLD,    "KE", "KENYA",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_KOREA_ROC,   APL9_MKKC,    "KR", "KOREA ROC", YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_KUWAIT,      ETSI3_WORLD,   "KW", "KUWAIT",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LATVIA,      ETSI13_WORLD,   "LV", "LATVIA",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LEBANON,     FCC3_WORLD,    "LB", "LEBANON",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LESOTHO, ETSI1_WORLD, "LS", "LESOTHO",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LIECHTENSTEIN, ETSI13_WORLD, "LI", "LIECHTENSTEIN",  YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LITHUANIA,   ETSI13_WORLD,   "LT", "LITHUANIA",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_LUXEMBOURG,  ETSI13_WORLD,   "LU", "LUXEMBOURG",     YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MACAU,       FCC3_WORLD,    "MO", "MACAU / MACAO",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MACEDONIA,   ETSI13_WORLD,   "MK", "MACEDONIA", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MALAWI, ETSI1_WORLD, "MW", "MALAWI",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MALAYSIA,    FCC11_WORLD,    "MY", "MALAYSIA",       YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MALDIVES, APL6_WORLD, "MV", "MALDIVES",       YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MALTA,       ETSI13_WORLD,   "MT", "MALTA",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MARSHALL_ISLANDS, FCC3_FCCA, "MH", "MARSHALL ISLANDS",       YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MARTINIQUE, ETSI13_WORLD, "MQ", "MARTINIQUE",       YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MAURITANIA, ETSI1_WORLD, "MR", "MAURITANA",       YES,  YES,  YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MAURITIUS,   ETSI13_WORLD,   "MU", "MAURITIUS",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MAYOTTE, ETSI1_WORLD, "YT", "MAYOTTE",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MEXICO,      FCC3_ETSIC,    "MX", "MEXICO",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MICRONESIA, FCC3_FCCA, "FM", "MICRONESIA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MOLDOVA, ETSI13_WORLD, "MD", "MOLDOVA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MONACO,      ETSI13_WORLD,   "MC", "MONACO",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MONGOLIA, FCC3_WORLD, "MN", "MONGOLIA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MONTENEGRO,  ETSI13_WORLD,   "ME", "MONTENEGRO",     YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MONTSERRAT,  ETSI13_WORLD,   "MS", "MONTSERRAT",     YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MOROCCO,     ETSI3_WORLD,    "MA", "MOROCCO",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_MYANMAR,  APL1_WORLD,    "MM", "MYANMAR",     YES,  YES, YES, YES, YES,  YES,  YES, YES, 7000 },
+    {CTRY_NAMIBIA, APL20_WORLD, "NA", "NAMIBIA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NEPAL,       APL23_WORLD,    "NP", "NEPAL",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NETHERLANDS, ETSI13_WORLD,   "NL", "NETHERLANDS",    YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NETHERLANDS_ANTILLES, ETSI13_WORLD, "AN", "NETHERLANDS ANTILLES", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NEW_CALEDONIA, ETSI13_WORLD, "NC", "NEW CALEDONIA", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NEW_ZEALAND, FCC3_ETSIC,    "NZ", "NEW ZEALAND",    YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NIGERIA, APL8_WORLD, "NG", "NIGERIA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NIUE, ETSI13_WORLD, "NU", "NIUE",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NORFOLK_ISLAND, FCC6_WORLD, "NF", "NORFOLK ISLAND",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_NORTHERN_MARIANA_ISLANDS, FCC3_FCCA, "MP", "MARIANA ISLANDS",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_NICARAGUA,   FCC3_FCCA,     "NI", "NICARAGUA",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_NORWAY,      ETSI1_WORLD,   "NO", "NORWAY",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_OMAN,        FCC3_WORLD,    "OM", "OMAN",           YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
@@ -637,42 +683,60 @@ static const COUNTRY_CODE_TO_ENUM_RD ahCmnAllCountries[] = {
     {CTRY_PARAGUAY,    FCC3_WORLD,    "PY", "PARAGUAY",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_PERU,        FCC3_WORLD,    "PE", "PERU",           YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_PHILIPPINES, FCC3_WORLD,    "PH", "PHILIPPINES",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_POLAND,      ETSI1_WORLD,   "PL", "POLAND",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_PORTUGAL,    ETSI1_WORLD,   "PT", "PORTUGAL",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_PUERTO_RICO, FCC1_FCCA,     "PR", "PUERTO RICO",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_QATAR,       APL1_WORLD,    "QA", "QATAR",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_ROMANIA,     ETSI1_WORLD,   "RO", "ROMANIA",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_RUSSIA,      ETSI8_WORLD,   "RU", "RUSSIA",         YES,  NO, NO,  YES, YES, YES, YES, NO, 7000 },
-    {CTRY_RWANDA,      APL1_WORLD,    "RW", "RWANDA",         YES,  NO, YES, YES, YES, YES,  YES, YES, 7000 },
-    {CTRY_SAUDI_ARABIA, FCC2_WORLD,   "SA", "SAUDI ARABIA",   YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_SERBIA,      ETSI1_WORLD,   "RS", "REPUBLIC OF SERBIA", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_MONTENEGRO,  ETSI1_WORLD,   "ME", "MONTENEGRO",     YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_POLAND,      ETSI13_WORLD,   "PL", "POLAND",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_PORTUGAL,    ETSI13_WORLD,   "PT", "PORTUGAL",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_PUERTO_RICO, FCC3_FCCA,     "PR", "PUERTO RICO",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_QATAR,      ETSI14_WORLD,    "QA", "QATAR",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_REUNION, ETSI1_WORLD, "RE", "REUNION",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_ROMANIA,     ETSI13_WORLD,   "RO", "ROMANIA",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_RUSSIA,      ETSI8_WORLD,   "RU", "RUSSIA",         YES,  YES, YES,  YES, YES, YES, YES, YES, 7000 },
+    {CTRY_RWANDA,      FCC3_WORLD,    "RW", "RWANDA",         YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_SAINT_BARTHELEMY, ETSI1_WORLD, "BL", "SAINT BARTHELEMY",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_HELENA_ASCENSION_AND_TRISTAN_DA_CUNHA, ETSI13_WORLD, "SH", "SAINT HELENA ASCENSION AND TRISTAN DA CUNHA",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_KITTS_AND_NEVIS, APL10_WORLD, "KN", "SAINT KITTS AND NEVIS",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_LUCIA, APL10_WORLD, "LC", "SAINT LUCIA",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_MARTIN, ETSI1_WORLD, "MF", "SAINT MARTIN",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_PIERRE_AND_MIQUELON, ETSI13_WORLD, "PM", "SAINT PIERRE AND MIQUELON",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAINT_VINCENT_AND_THE_GRENADIENS, ETSI13_WORLD, "VC", "VINCENT AND THE GRENADIENS",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAMOA, ETSI1_WORLD, "WS", "SAMOA",    YES, YES, YES, YES, YES, YES, YES, NO, 7000 },
+    {CTRY_SAN_MARINO, FCC3_FCCA, "SM", "SAN MARINO",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAO_TOME_AND_PRINCIPE, FCC3_WORLD, "ST", "SAO TOME AND PRINCIPE",    YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SAUDI_ARABIA, ETSI15_WORLD,   "SA", "SAUDI ARABIA",   YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SENEGAL, FCC13_WORLD, "SN", "SENEGAL",   YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SERBIA,      ETSI13_WORLD,   "RS", "REPUBLIC OF SERBIA", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_SINGAPORE,   FCC3_WORLD,    "SG", "SINGAPORE",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SLOVAKIA,    ETSI1_WORLD,   "SK", "SLOVAKIA",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SLOVENIA,    ETSI1_WORLD,   "SI", "SLOVENIA",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SOUTH_AFRICA, FCC3_WORLD,   "ZA", "SOUTH AFRICA",   YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SPAIN,       ETSI1_WORLD,   "ES", "SPAIN",          YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SRI_LANKA,   FCC3_WORLD,    "LK", "SRI LANKA",      YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_SWEDEN,      ETSI1_WORLD,   "SE", "SWEDEN",         YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SWITZERLAND, ETSI1_WORLD,   "CH", "SWITZERLAND",    YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_SYRIA,       NULL1_WORLD,   "SY", "SYRIAN ARAB REPUBLIC", YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
-    {CTRY_TAIWAN,      APL7_FCCA,     "TW", "TAIWAN",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SINT_MAARTEN,   ETSI13_WORLD,    "SX", "SINT MAARTEN",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SLOVAKIA,    ETSI13_WORLD,   "SK", "SLOVAKIA",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SLOVENIA,    ETSI13_WORLD,   "SI", "SLOVENIA",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SOUTH_AFRICA, FCC3_WORLD,   "ZA", "SOUTH AFRICA",   YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SPAIN,       ETSI13_WORLD,   "ES", "SPAIN",          YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SURINAME, ETSI1_WORLD, "SR", "SURINAME",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SVALBARD_AND_JAN_MAYEN, FCC6_WORLD, "SJ", "SVALBARD AND JAN MAYEN",      YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SRI_LANKA,   FCC3_WORLD,    "LK", "SRI LANKA",      YES,  YES, YES, YES, YES, YES,  NO, NO, 7000 },
+    {CTRY_SWEDEN,      ETSI13_WORLD,   "SE", "SWEDEN",         YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_SWITZERLAND, ETSI13_WORLD,   "CH", "SWITZERLAND",    YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_TAIWAN,      FCC3_FCCA,     "TW", "TAIWAN, PROVINCE OF CHINA",         YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_TANZANIA,    APL1_WORLD,    "TZ", "TANZANIA",       YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_THAILAND,    FCC3_WORLD,    "TH", "THAILAND",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_TRINIDAD_Y_TOBAGO, FCC3_WORLD, "TT", "TRINIDAD AND TOBAGO", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_TUNISIA,     ETSI3_WORLD,   "TN", "TUNISIA",        YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_TURKEY,      ETSI3_WORLD,   "TR", "TURKEY",         YES,  NO, YES, YES, YES, YES,  NO, NO, 7000 },
-    {CTRY_UGANDA,      FCC3_WORLD,    "UG", "UGANDA",         YES,  NO, NO, YES, YES, YES,  YES, YES, 7000 },
-    {CTRY_UKRAINE,     ETSI9_WORLD,   "UA", "UKRAINE",        YES,  NO, NO,  YES, YES, YES,  YES, NO, 7000 },
-    {CTRY_UAE,         ETSI1_WORLD, "AE", "UNITED ARAB EMIRATES", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_UNITED_KINGDOM, ETSI1_WORLD, "GB", "UNITED KINGDOM", YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_UNITED_STATES, FCC3_FCCA,   "US", "UNITED STATES",  YES, YES, YES, YES, YES, YES, YES, YES, 5825 },
-    {CTRY_UNITED_STATES2, FCC6_FCCA,  "US", "UNITED STATES2",  YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_UNITED_STATES_FCC49, FCC4_FCCA,   "PS", "UNITED STATES (PUBLIC SAFETY)",  YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_URUGUAY,     FCC3_WORLD,    "UY", "URUGUAY",        YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_UZBEKISTAN,  FCC3_FCCA,     "UZ", "UZBEKISTAN",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_VENEZUELA,   FCC1_WORLD,    "VE", "VENEZUELA",      YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
-    {CTRY_VIET_NAM,    ETSI3_WORLD,   "VN", "VIET NAM",       YES,  NO, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_THAILAND,    FCC3_WORLD,    "TH", "THAILAND",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_TOGO, ETSI1_WORLD, "TG", "TOGO",       YES, YES, YES, YES, YES, YES, YES, NO, 7000 },
+    {CTRY_TRINIDAD_Y_TOBAGO, FCC3_WORLD, "TT", "TRINIDAD AND TOBAGO", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_TUNISIA,     ETSI3_WORLD,   "TN", "TUNISIA",        YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_TURKEY,      ETSI13_WORLD,   "TR", "TURKEY",         YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_TURKS_AND_CAICOS, FCC3_WORLD, "TC", "TURKS AND CAICOS",       YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_UGANDA,      FCC3_WORLD,    "UG", "UGANDA",         YES,  YES, YES, YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_UKRAINE,     ETSI9_WORLD,   "UA", "UKRAINE",        YES,  YES, YES,  YES, YES, YES,  YES, YES, 7000 },
+    {CTRY_UAE,         FCC3_WORLD, "AE", "UNITED ARAB EMIRATES", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_UNITED_KINGDOM, ETSI13_WORLD, "GB", "UNITED KINGDOM", YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_UNITED_STATES, FCC8_FCCA,   "US", "UNITED STATES",  YES, YES, YES, YES, YES, YES, YES, YES, 5825 },
+    {CTRY_UNITED_STATES_MINOR_OUTLYING_ISLANDS, FCC3_FCCA,   "UM", "UNITED STATES MINOR OUTLYING ISLANDS",  YES, YES, YES, YES, YES, YES, YES, YES, 5825 },
+    {CTRY_URUGUAY,     FCC2_WORLD,    "UY", "URUGUAY",        YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_UZBEKISTAN,  ETSI3_WORLD,     "UZ", "UZBEKISTAN",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_VANUATU, FCC3_WORLD, "VU", "VANUATU",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_VENEZUELA,   FCC2_ETSIC,    "VE", "VENEZUELA",      YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_VIET_NAM,    FCC3_WORLD,   "VN", "VIET NAM",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_VIRGIN_ISLANDS_BRITISH,    ETSI13_WORLD,   "VG", "VIRGIN ISLANDS BRITISH",       YES,  YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_VIRGIN_ISLANDS, FCC3_FCCA, "VI", "VIRGIN ISLANDS (U.S)",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
+    {CTRY_WALLIS_AND_FUTUNA, ETSI1_WORLD, "WF", "WALLIS AND FUTUNA",     YES, YES, YES, YES, YES, YES, YES, YES, 7000 },
     {CTRY_YEMEN,       NULL1_WORLD,   "YE", "YEMEN",          YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 },
     {CTRY_ZIMBABWE,    ETSI1_WORLD,   "ZW", "ZIMBABWE",       YES,  NO, YES, YES, YES,  NO,  NO, NO, 7000 }
 };
@@ -1253,7 +1317,7 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
-	{APL8, ETSI, NO_DFS, NO_PSCAN, DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
+	{APL8, FCC, DFS_ETSI, NO_PSCAN, DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
 		BM(F6_5260_5320, F4_5745_5825, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T2_5290_5290, T2_5760_5800, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T1_5280_5280, T1_5765_5805, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
@@ -1262,7 +1326,7 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
-	{APL9, ETSI, DFS_ETSI, PSCAN_ETSI, DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
+	{APL9, MKK, DFS_ETSI, PSCAN_ETSI, DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
 		BM(F9_5180_5240, F2_5260_5320, F1_5500_5620, F3_5745_5805, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T3_5290_5290, T5_5760_5800, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T1_5540_5660, T6_5765_5805, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
@@ -1280,7 +1344,7 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
-	{APL11, ETSI, DFS_ETSI, PSCAN_ETSI , DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
+	{APL11, ETSI, DFS_FCC3, PSCAN_ETSI , DISALLOW_ADHOC_11A|DISALLOW_ADHOC_11A_TURB,
 		BM(F9_5180_5240, F2_5260_5320, F5_5500_5700, F7_5745_5825, F1_5845_5865, -1, -1, -1, -1, -1, -1, -1),
 		BM(T3_5290_5290, T5_5760_5800, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T1_5540_5660, T6_5765_5805, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
@@ -1307,6 +1371,68 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
+	{APL17, FCC, NO_DFS, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{APL14, MKK, DFS_FCC3, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{APL15, FCC, NO_DFS, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+	{APL19, FCC, DFS_FCC3, PSCAN_ETSI, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{APL20, ETSI, DFS_ETSI, PSCAN_ETSI, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{APL23, ETSI, NO_DFS, PSCAN_ETSI, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{APL24, ETSI, DFS_ETSI, PSCAN_ETSI, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
 	{ETSI1, ETSI, DFS_ETSI, PSCAN_ETSI, DISALLOW_ADHOC_11A | DISALLOW_ADHOC_11A_TURB,
 		BM(F2_5180_5240, F2_5260_5320, F2_5500_5700, -1, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T1_5210_5290, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
@@ -1440,6 +1566,51 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
+	{FCC8, FCC, DFS_FCC3, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{FCC11, FCC, DFS_FCC3, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{FCC13, FCC, NO_DFS, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{FCC14, FCC, NO_DFS, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
+	{KRRA, KOR, NO_DFS, PSCAN_FCC, NO_REQ,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO
+	},
+
 	{MKK1, MKK, DFS_MKK4, PSCAN_MKK1 | PSCAN_MKK3, DISALLOW_ADHOC_11A_TURB,
 		BM(F1_5170_5230, F10_5180_5240, F7_5260_5320, F4_5500_5700, -1, -1, -1, -1, -1, -1, -1, -1),
 		BM(T7_5210_5210, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1),
@@ -1588,6 +1759,15 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BMZERO
 	},
 
+	{MKK17, MKK, DFS_MKK4, PSCAN_MKK1 | PSCAN_MKK3, DISALLOW_ADHOC_11A_TURB,
+		BM(F1_4915_4925, F1_4935_4945, F2_4920_4980, F1_5040_5040, F2_5040_5080, F1_5055_5055, F2_5170_5230, F4_5180_5240, F2_5260_5320, -1, -1, -1),
+		BMZERO,
+		BMZERO,
+		BMZERO,
+		BMZERO,
+		CHAN_TURBO_G_BMZERO
+	},
+
 	/*=== 2 GHz ===*/
 
 	/* Defined here to use when 2G channels are authorised for country K2 */
@@ -1627,6 +1807,24 @@ static const REG_DOMAIN ahCmnRegDomains[] = {
 		CHAN_TURBO_G_BM(T2_2437_2437, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1)
 	},
 
+	{ETSID, ETSI, NO_DFS, PSCAN_ETSI, NO_REQ,
+		CHAN_11A_BMZERO
+		CHAN_11A_BMZERO
+		CHAN_11A_BMZERO
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO
+	},
+
+	{ETSI10, ETSI, DFS_ETSI, PSCAN_ETSI, NO_REQ,
+		CHAN_11A_BMZERO
+		CHAN_11A_BMZERO
+		CHAN_11A_BMZERO
+		BMNOTZERO,
+		BMNOTZERO,
+		BMNOTZERO
+	},
+
 	{FCCA, FCC, NO_DFS, NO_PSCAN, NO_REQ,
 		CHAN_11A_BMZERO
 			CHAN_11A_BMZERO
-- 
2.14.5

